sktime/regression/deep_learning/cnn.py

```diff
@@ -162,8 +162,7 @@ def _fit(self, X, y):
         -------
         self : object
         """
-        if self.callbacks is None:
-            self._callbacks = []
+        self.callbacks = self.callbacks or []
```

Comment:
The problem is here where accidental overwriting is taking place, we should keep the structure the same as it was for classification estimators.